### PR TITLE
support toc href and return toc parser errors

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

File: src/docfx/Errors.cs [hunk 1]
```diff
@@ -24,6 +24,12 @@ public static DocfxException InvalidConfig(string configPath, Exception e)
         public static DocfxException CircularReference<T>(T filePath, IEnumerable<T> dependencyChain)
             => new DocfxException(ReportLevel.Error, "circular-reference", $"Found circular reference: {string.Join(" --> ", dependencyChain.Select(file => $"'{file}'"))} --> '{filePath}'", filePath.ToString());
 
+        public static DocfxException InvalidTopicHref(string topicHref)
+            => new DocfxException(ReportLevel.Error, "invalid-topc-href", $"The topic href {topicHref} can only reference to a local file or absolute path");
```
Comment (yufeih) [inline src/docfx/Errors.cs]: >{topicHref} [](start = 91, length = 11)

‘{topicHref}’

File: src/docfx/build/toc/BuildTableOfContents.cs [hunk 2]
```diff
@@ -73,7 +73,7 @@ private static Task BuildTocMap(Document fileToBuild, TableOfContentsMapBuilder
             var errors = new List<DocfxException>();
             var referencedDocuments = new List<Document>();
             var referencedTocs = new List<Document>();
-            var tocViewModel = TableOfContentsParser.Load(
+            var (tocViewModel, loadErros) = TableOfContentsParser.Load(
```
Comment (yufeih) [inline src/docfx/build/toc/BuildTableOfContents.cs]: >loadErros [](start = 31, length = 9)

errors?

File: src/docfx/build/toc/parser/TableOfContentsParser.cs [hunk 3]
```diff
@@ -15,8 +16,11 @@ internal static class TableOfContentsParser
 
         public delegate (string content, Document file) ResolveContent(Document relativeTo, string href, bool isInclusion);
 
-        public static List<TableOfContentsItem> Load(string tocContent, Document filePath, ResolveContent resolveContent = null, ResolveHref resolveHref = null)
-            => LoadInputModelItems(tocContent, filePath, filePath, resolveContent, resolveHref)?.Select(r => TableOfContentsInputItem.ToTableOfContentsModel(r)).ToList();
+        public static (List<TableOfContentsItem> model, List<DocfxException> errors) Load(string tocContent, Document filePath, ResolveContent resolveContent, ResolveHref resolveHref)
```
Comment (yufeih) [inline src/docfx/build/toc/parser/TableOfContentsParser.cs]: >errors [](start = 77, length = 6)

errors first?

File: src/docfx/build/toc/BuildTableOfContents.cs (hunk 2 above)
Comment (OsmondJiang) [reply src/docfx/build/toc/BuildTableOfContents.cs]: loadErrors, typo

---
In reply to: [197333863](https://github.com/dotnet/docfx/pull/2918#discussion_r197333863) [](ancestors = 197333863)

**Comment by OsmondJiang** [comment]
#2807

**Pull request merged**